File: compiler/Jvm/src/com/stratos/model/Statement/OperationExpressionStatement.java
package com.stratos.model.Statement;

import com.stratos.model.Node;

import java.util.List;

// math expression ie
// add, minus, divide, subtract , modulas.
// presendence BODMAS (brackats, Of , division , multiplication , addition , subtraction
public class OperationExpressionStatement {
    Node leftValue;
    Node operationValue;
    Node rightValue;

    boolean complexStatement;
    List<OperationExpressionStatement> operationExpressionStatementList;

    public boolean isComplexStatement() {
        return complexStatement;
    }

    public void setComplexStatement(boolean complexStatement) {
        this.complexStatement = complexStatement;
    }

    public List<OperationExpressionStatement> getOperationExpressionStatementList() {
        return operationExpressionStatementList;
    }

    public void setOperationExpressionStatementList(List<OperationExpressionStatement> operationExpressionStatementList) {
        this.operationExpressionStatementList = operationExpressionStatementList;
    }

    public Node getLeftValue() {
        return leftValue;
    }

    public void setLeftValue(Node leftValue) {
        this.leftValue = leftValue;
    }

    public Node getOperationValue() {
        return operationValue;
    }

    public void setOperationValue(Node operationValue) {
        this.operationValue = operationValue;
    }

    public Node getRightValue() {
        return rightValue;
    }

    public void setRightValue(Node rightValue) {
        this.rightValue = rightValue;
    }
}

File: compiler/Jvm/src/com/stratos/model/Statement/OperationExpressionStatement.kt
package com.stratos.model.Statement

import com.stratos.model.Node

// math expression ie
// add, minus, divide, subtract , modulas.
// presendence BODMAS (brackats, Of , division , multiplication , addition , subtraction
class OperationExpressionStatement : Statement {
    var returnValue: Node?
    var leftValue: Node
    var operationValue: Node
    var rightValue: Node
    var isComplexStatement = false
    var operationExpressionStatementList: List<OperationExpressionStatement>? = null

    constructor(returnValue: Node?, leftValue: Node, operationValue: Node, rightValue: Node) {
        this.returnValue = returnValue
        this.leftValue = leftValue
        this.operationValue = operationValue
        this.rightValue = rightValue
    }

    constructor(returnValue: Node, leftValue: Node, operationValue: Node, rightValue: Node, complexStatement: Boolean, operationExpressionStatementList: List<OperationExpressionStatement>?) {
        this.returnValue = returnValue
        this.leftValue = leftValue
        this.operationValue = operationValue
        this.rightValue = rightValue
        isComplexStatement = complexStatement
        this.operationExpressionStatementList = operationExpressionStatementList
    }

}